aiohttp/connector.py

```diff
@@ -464,11 +491,24 @@ def _create_connection(self, req):
 
         for hinfo in hosts:
             try:
-                return (yield from self._loop.create_connection(
-                    self._factory, hinfo['host'], hinfo['port'],
+                host = hinfo['host']
+                port = hinfo['port']
+                conn = yield from self._loop.create_connection(
+                    self._factory, host, port,
                     ssl=sslcontext, family=hinfo['family'],
                     proto=hinfo['proto'], flags=hinfo['flags'],
-                    server_hostname=hinfo['hostname'] if sslcontext else None))
+                    server_hostname=hinfo['hostname'] if sslcontext else None)
+                if self._verify_fingerprint:
+                    sock = conn[0]._sock
+                    # gives DER-encoded cert as a sequence of bytes (or None)
+                    cert = sock.getpeercert(binary_form=True)
```

Comment:
@asvetlov Do these two lines (502 and 504) look good to you? The `conn[0]._sock` makes me feel the need to check because of the private member access. As for `sock.getpeercert`, if sock is a regular `socket.socket` rather than an `SSLSocket` this will cause `AttributeError`, but I think that would only happen if the user tries to use `verify_fingerprint` a non-SSL connection. I can code this more defensively though if that would be better.